Gordon360/AuthorizationFilters/StateYourBusiness.cs

```diff
@@ -503,6 +542,26 @@ private bool canUpdate(string resource)
                     }
                 case Resource.STUDENT:
                     return false; // No one should be able to update a student through this API
+                case Resource.HOUSING:
+                    {
+                        // The user must be a student and an editor to update the application
+                        if (user_position == Position.STUDENT)
+                        {
+                            var housingService = new HousingService(new UnitOfWork());
+                            var sess_cde = Helpers.GetCurrentSession().ToString();
+                            int? applicationID = housingService.GetApplicationID(user_id, sess_cde);
+                            if (applicationID != null)
+                            {
+                                var editorID = housingService.GetEditorID(applicationID);
+                                var is_editor = editorID == user_id;
+                                if (is_editor)
+                                    return true;
+                            }
+                            return false;
```

Comment:
We will have to remember to update things like this as soon as we remake the database tables because the `user_id` will become `username` and `editorID` will become `editorUsername`